xen-ocaml/core/build.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh -ex
+
+export PKG_CONFIG_PATH=`opam config var prefix`/lib/pkgconfig
+PKG_CONFIG_DEPS="openlibm libminios-xen >= 0.5"
+pkg-config --print-errors --exists ${PKG_CONFIG_DEPS} || exit 1
+
+case `uname -m` in
+armv*)
+  ARCH_CFLAGS="-DTARGET_arm"
+  m_header=m.arm.h
+ ;;
+*)
+  ARCH_CFLAGS="-DTARGET_amd64 -D__x86_64__ -momit-leaf-frame-pointer -mfancy-math-387"
+  m_header=m.x86_64.h
+  ;;
+esac
+
+# This extra flag only needed for gcc 4.8+
+# TODO we only need this for minios and not applications I think
+GCC_MVER2=`gcc -dumpversion | cut -f2 -d.`
+if [ $GCC_MVER2 -ge 8 ]; then
+  EXTRA_CFLAGS=-fno-tree-loop-distribute-patterns
+fi
```

Comment:
As the TODO says, it shouldn't be needed outside of Mini-OS (unless mirage-xen is redefining `memcpy` or something). According to http://sourceware.org/ml/libc-alpha/2013-04/msg00128.html we should be using `__attribute__ ((optimize ("no-tree-loop-distribute-patterns")))` on the individual functions anyway, which makes sense.